File: 156_Problem_1.py
# code_report Solution
# Problem Link: https://leetcode.com/contest/weekly-contest-156/problems/unique-number-of-occurrences/

def group(l): 
    return [list(x) for _, x in itertools.groupby(l)]

def freq(l):  
    return [len(x) for x in group(sorted(l))]

def uniqueOccurrences(self, arr: List[int]) -> bool:
    return max(freq(freq(arr))) <= 1

File: 156_Problem_1.hs
-- code_report Solution
-- Problem Link (Contest):  https://leetcode.com/contest/weekly-contest-156/problems/unique-number-of-occurrences/
-- Problem Link (Practice): https://leetcode.com/problems/unique-number-of-occurrences/

import Data.List

freq :: [Int] -> [Int]
freq = map length 
     . group 
     . sort

uniqueOccurrences :: [Int] -> Bool
uniqueOccurrences = (<=1) 
                  . maximum 
                  . freq
                  . freq
